package.json

```
@@ -4,6 +4,10 @@
   "version": "3.0.2",
   "description": "The URI implementation that is used by VS Code and its extensions",
   "main": "./lib/umd/index.js",
+  "exports": {
+    "import": "./lib/esm/index.mjs",
+    "require": "./lib/umd/index.js"
```

This also needs `exports.types` for the `node16` / `nodenext` module resolution in the TypeScript compiler options.

```suggestion
    "require": "./lib/umd/index.js",
    "types": "./lib/umd/index.d.ts"
```

Apart from that, great addition!

---

I believe that I didn't need that when testing with https://github.com/typescript-language-server/typescript-language-server/ but I will do some more testing later. I'm already using `"moduleResolution": "Node16"` there.

Also I wonder whether "types" is specific to only some of the following: Node, Webpack, Typescript...

---

You can try it by creating the following files:

`package.json`

```json
{
  "type": "module",
  "dependencies": {
    "typescript": "^4.8.2",
    "vscode-uri": "^3.0.3"
  }
}
```

`tsconfig.json`

```json
{
  "compilerOptions": {
    // implies "moduleResolution": "node16",
    "module": "node16",
    "noEmit": true,
    "strict": true,
  }
}
```

`foo.ts`

```ts
import {URI} from 'vscode-uri'

console.log(URI.parse('file:///foo.ts'))
```

Then make the modifications in `node_modules/vscode-uri/package.json`. To test, simply run `npx tsc` or `yarn tsc`. You’ll find out `exports.types` is indeed required.

Unfortunately a lot of existing packages aren’t properly configured yet. I’m glad to see someone making this effort. :smile:

---

Thanks. Tested and confirmed your findings.